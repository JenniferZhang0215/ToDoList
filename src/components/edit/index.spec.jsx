import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Edit from "./index";

const props = {
  onCancel: jest.fn(),
  onSave: jest.fn(),
  defaultValue: "fake default value",
};

configure({ adapter: new Adapter() });

describe("<Edit />", () => {
  const wrapper = shallow(<Edit {...props} />);

  it("should render default snapshot", () => {
    expect(wrapper).toMatchSnapshot("default");
  });

  describe("edit input", () => {
    const input = wrapper.findWhere((node) => node.key() === "edit-input");
    it("should call handleEditItem function when edit input is changed", () => {
      const e = { target: { value: "edit input" } };
      input.simulate("change", e);
      expect(wrapper.state("editText")).toEqual(e.target.value);
    });
    it("should pass defaultValue to input correctly", () => {
      expect(input.props().defaultValue).toEqual(props.defaultValue);
    });
  });

  describe("cancel button", () => {
    const button = wrapper.findWhere((node) => node.key() === "cancel-btn");
    it("should pass onCancel correctly", () => {
      expect(button.props().onClick).toEqual(props.onCancel);
    });
  });

  describe("save button", () => {
    it("should call onSave function if the save button is clicked", () => {
      const button = wrapper.findWhere((node) => node.key() === "save-btn");
      const mockEditText = wrapper.state("editText");
      button.simulate("click");
      expect(props.onSave).toHaveBeenCalledWith(mockEditText);
    });
  });
});
