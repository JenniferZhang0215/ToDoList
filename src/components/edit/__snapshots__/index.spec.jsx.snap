// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edit /> should render default snapshot: default 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Edit
    defaultValue="fake default value"
    onCancel={[MockFunction]}
    onSave={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <input
          defaultValue="fake default value"
          onChange={[Function]}
        />,
        <button
          className="btn-sm m-1"
          onClick={[MockFunction]}
        >
          Cancel
        </button>,
        <button
          className="btn-success btn-sm m-1"
          onClick={[Function]}
        >
          Save
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "edit-input",
        "nodeType": "host",
        "props": Object {
          "defaultValue": "fake default value",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": "cancel-btn",
        "nodeType": "host",
        "props": Object {
          "children": "Cancel",
          "className": "btn-sm m-1",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Cancel",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "save-btn",
        "nodeType": "host",
        "props": Object {
          "children": "Save",
          "className": "btn-success btn-sm m-1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Save",
        "type": "button",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <input
            defaultValue="fake default value"
            onChange={[Function]}
          />,
          <button
            className="btn-sm m-1"
            onClick={[MockFunction]}
          >
            Cancel
          </button>,
          <button
            className="btn-success btn-sm m-1"
            onClick={[Function]}
          >
            Save
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "edit-input",
          "nodeType": "host",
          "props": Object {
            "defaultValue": "fake default value",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": "cancel-btn",
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
            "className": "btn-sm m-1",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "save-btn",
          "nodeType": "host",
          "props": Object {
            "children": "Save",
            "className": "btn-success btn-sm m-1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Save",
          "type": "button",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
